File: src/main/java/me/mrlion/lionplugin/LionPlugin.java
package me.mrlion.lionplugin;

import org.bukkit.plugin.java.JavaPlugin;
import me.mrlion.lionplugin.util.CommandRegister;
import me.mrlion.lionplugin.util.ListenerRegister;

public class LionPlugin extends JavaPlugin {

    @Override
    public void onEnable() {
        try {
            CommandRegister.registerCommands(this);
            ListenerRegister.registerListeners(this);
            saveDefaultConfig();
            getLogger().info("LionPlugin has been enabled!");
        } catch (Exception e) {
            getLogger().severe("Error occurred while enabling LionPlugin:");
            e.printStackTrace();
            getServer().getPluginManager().disablePlugin(this);
        }
    }
    @Override
    public void onDisable() {
        getLogger().info("LionPlugin has been disabled!");
    }
}

File: src/main/java/me/mrlion/lionplugin/LionPlugin.kt
package me.mrlion.lionplugin

import me.mrlion.lionplugin.commands.*
import me.mrlion.lionplugin.listeners.BlockedCommandsListener
import me.mrlion.lionplugin.listeners.InvalidCommandListener
import me.mrlion.lionplugin.listeners.JoinQuitListener
import me.mrlion.lionplugin.listeners.SitListener
import org.bukkit.plugin.java.JavaPlugin
import java.util.*

class LionPlugin : JavaPlugin() {
    var vanishedPlayers: MutableSet<UUID>? = null
        private set

    override fun onEnable() {
        vanishedPlayers = HashSet()

        Objects.requireNonNull(getCommand("fly"))?.setExecutor(FlyCommand(this))
        Objects.requireNonNull(getCommand("heal"))?.setExecutor(HealCommand(this))
        Objects.requireNonNull(getCommand("msg"))?.setExecutor(MsgCommand(this))
        Objects.requireNonNull(getCommand("lreload"))?.setExecutor(ReloadCommand(this))
        Objects.requireNonNull(getCommand("ping"))?.setExecutor(PingCommand(this))
        Objects.requireNonNull(getCommand("vanish"))?.setExecutor(VanishCommand(this))

        server.pluginManager.registerEvents(JoinQuitListener(this), this)
        server.pluginManager.registerEvents(BlockedCommandsListener(this), this)
        server.pluginManager.registerEvents(SitListener(this), this)
        server.pluginManager.registerEvents(InvalidCommandListener(this), this)

        logger.info("LionPlugin has been enabled!")
        saveDefaultConfig()
        reloadConfig()
    }

    override fun onDisable() {
        logger.info("LionPlugin has been disabled!")
    }
}
